File: guides/micronaut-microservices-distributed-tracing-xray/bookrecommendation/java/src/main/java/example/micronaut/BookCatalogueClient.java
//tag::packageandimports[]
package example.micronaut;

import io.micronaut.http.annotation.Get;
import io.micronaut.http.client.annotation.Client;
import io.micronaut.retry.annotation.Recoverable;
import org.reactivestreams.Publisher;
//end::packageandimports[]

//tag::harcoded[]
@Client(id = "bookcatalogue") // <1>
@Recoverable(api = BookCatalogueOperations.class)
//end::harcoded[]
//tag::clazz[]
public interface BookCatalogueClient extends BookCatalogueOperations {

    @Get("/books")
    Publisher<Book> findAll();
}
//end::clazz[]

File: guides/micronaut-oauth2-client-credentials-auth0/bookrecommendation/kotlin/src/main/kotlin/example/micronaut/BookCatalogueClient.kt
package example.micronaut

import io.micronaut.http.annotation.Get
import io.micronaut.http.client.annotation.Client
import io.micronaut.retry.annotation.Recoverable
import org.reactivestreams.Publisher

@Client(id = "bookcatalogue")
@Recoverable(api = BookCatalogueOperations::class)
interface BookCatalogueClient : BookCatalogueOperations {

    @Get("/books")
    override fun findAll(): Publisher<Book>
}
